shell/browser/api/electron_api_web_contents.cc

```diff
@@ -1386,6 +1389,32 @@ void WebContents::FindReply(content::WebContents* web_contents,
   Emit("found-in-page", result.GetHandle());
 }
 
+void WebContents::RequestExclusivePointerAccess(
+    content::WebContents* web_contents,
+    bool user_gesture,
+    bool last_unlocked_by_target,
+    bool allowed) {
+  if (allowed) {
+    exclusive_access_manager_->mouse_lock_controller()->RequestToLockMouse(
+        web_contents, user_gesture, last_unlocked_by_target);
+  }
```

Comment:
Shouldn't the not allowed case from `setPermissionRequestHandler` still continue to fire `web_contents->GotResponseToLockMouseRequest(blink::mojom::PointerLockResult::kPermissionDenied);` so that any active requests get cancelled ?

Reply:
good point, let's do that!